Errors when running npm run dev -- --host

Ran:
`npm run dev -- --host example.test`

Saw:
```
> testsite.test@1.0.0 dev /home/mn/git/portfolio
> webpack-dev-server --env=dev --mode development --open "--host" "example.test"

/home/mn/git/portfolio/node_modules/webpack-dev-server/lib/utils/findPort.js:21
      throw err;
      ^

Error: getaddrinfo ENOTFOUND example.test
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:58:26)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.test"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the testsite.test@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mn/.npm/_logs/2019-07-08T03_41_26_325Z-debug.log
```

Logs:
```
 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'run',
1 verbose cli   'dev',
1 verbose cli   '--',
1 verbose cli   '--host',
1 verbose cli   'example.test' ]
2 info using npm@6.9.0
3 info using node@v11.15.0
4 verbose run-script [ 'predev', 'dev', 'postdev' ]
5 info lifecycle testsite.test@1.0.0~predev: testsite.test@1.0.0
6 info lifecycle testsite.test@1.0.0~dev: testsite.test@1.0.0
7 verbose lifecycle testsite.test@1.0.0~dev: unsafe-perm in lifecycle true
8 verbose lifecycle testsite.test@1.0.0~dev: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/mn/git/por>
9 verbose lifecycle testsite.test@1.0.0~dev: CWD: /home/mn/git/portfolio
10 silly lifecycle testsite.test@1.0.0~dev: Args: [ '-c',
10 silly lifecycle   'webpack-dev-server --env=dev --mode development --open "--host" "example.test"' ]
11 silly lifecycle testsite.test@1.0.0~dev: Returned: code: 1  signal: null
12 info lifecycle testsite.test@1.0.0~dev: Failed to exec dev script
13 verbose stack Error: testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.test"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:193:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:193:13)
13 verbose stack     at maybeClose (internal/child_process.js:999:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:266:5)
14 verbose pkgid testsite.test@1.0.0
15 verbose cwd /home/mn/git/portfolio
16 verbose Linux 5.1.4-arch1-1-ARCH
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "dev" "--" "--host" "example.test"
18 verbose node v11.15.0
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.test"`
22 error Exit status 1
23 error Failed at the testsite.test@1.0.0 dev script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

Another iteration, ran:
`npm run dev -- --host example.com`

Saw:
```
> testsite.test@1.0.0 dev /home/mn/git/portfolio
> webpack-dev-server --env=dev --mode development --open "--host" "example.com"

/home/mn/git/portfolio/node_modules/webpack-dev-server/lib/utils/findPort.js:21
      throw err;
      ^

Error: listen EADDRNOTAVAIL: address not available 93.184.216.34:8080
    at Server.setupListenHandle [as _listen2] (net.js:1242:19)
    at listenInCluster (net.js:1307:12)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1446:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:63:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.com"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the testsite.test@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mn/.npm/_logs/2019-07-08T03_54_24_813Z-debug.log
[mn@SapphireZen dist]$ nano /home/mn/.npm/_logs/2019-07-08T03_54_24_813Z-debug.log
```

Logs:
```
 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'run',
1 verbose cli   'dev',
1 verbose cli   '--',
1 verbose cli   '--host',
1 verbose cli   'example.com' ]
2 info using npm@6.9.0
3 info using node@v11.15.0
4 verbose run-script [ 'predev', 'dev', 'postdev' ]
5 info lifecycle testsite.test@1.0.0~predev: testsite.test@1.0.0
6 info lifecycle testsite.test@1.0.0~dev: testsite.test@1.0.0
7 verbose lifecycle testsite.test@1.0.0~dev: unsafe-perm in lifecycle true
8 verbose lifecycle testsite.test@1.0.0~dev: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/mn/git/por>
9 verbose lifecycle testsite.test@1.0.0~dev: CWD: /home/mn/git/portfolio
10 silly lifecycle testsite.test@1.0.0~dev: Args: [ '-c',
10 silly lifecycle   'webpack-dev-server --env=dev --mode development --open "--host" "example.com"' ]
11 silly lifecycle testsite.test@1.0.0~dev: Returned: code: 1  signal: null
12 info lifecycle testsite.test@1.0.0~dev: Failed to exec dev script
13 verbose stack Error: testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.com"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:193:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:193:13)
13 verbose stack     at maybeClose (internal/child_process.js:999:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:266:5)
14 verbose pkgid testsite.test@1.0.0
15 verbose cwd /home/mn/git/portfolio/public/dist
16 verbose Linux 5.1.4-arch1-1-ARCH
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "dev" "--" "--host" "example.com"
18 verbose node v11.15.0
19 verbose npm  v6.9.0
20 error code ELIFECYCLE
21 error errno 1
22 error testsite.test@1.0.0 dev: `webpack-dev-server --env=dev --mode development --open "--host" "example.com"`
22 error Exit status 1
23 error Failed at the testsite.test@1.0.0 dev script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```
Antergos
5.1.4-arch1-1-ARCH #1 SMP PREEMPT Wed May 22 08:06:56 UTC 2019 x86_64 GNU/Linux

--- a/README.md
+++ b/README.md
@@ -41,6 +41,8 @@ ___webpack.prod.___ - Production bundling rules
 ***npm run dev -- --host [hostname]***  
  - including a [hostname] will launch the Webpack dev server with the desired hostname in the browser  
  - Syntax: `npm run dev -- --host example.test`
+ - the inclusion of a hostname implys that your DNS is properly configured for that hostname and the top-level domain. Example, adding the following to your hosts file: `127.0.0.1 example.test`
+ - if you would like to control an entire top-level domain such as `.test` on your device, I'd recommend seeing this tutorial for DNSmasq [USE DNSMASQ INSTEAD OF /ETC/HOSTS](https://www.stevenrombauts.be/2018/01/use-dnsmasq-instead-of-etc-hosts/). This is heavyly geared toward macOS, but should work for most UNIX-like systems. Windows has alternatives.
 
 ***npm start***  
  - this will build using webpack and the bundle.js file will be in /public/dist
@@ -63,3 +65,6 @@ Revolve around essential setup and functionality, npm, react, docker, webpack, e
 [Made Nav Dynamic PR-6](https://github.com/jfasanobc/testsite.test/pull/6) - Made the UL and LI elements of the navigation dynamically loaded using ReactJS.  
 [Updated Readme PR-7](https://github.com/jfasanobc/testsite.test/pull/7) - Updated README.md with updated documentation and typing mistakes/markdown syntax errors.  
 [Added Jest Testing PR-9](https://github.com/jfasanobc/testsite.test/pull/9) - Added Jest testing to project with first initial test on the header component with snapshots included  
+
+***July 8, 2019***  
+[Resolved Issue #10 PR-](https://github.com/jfasanobc/testsite.test/pull/9) - Clarified how to use --host flag with npm run dev CLI command in README.  

--- a/package.json
+++ b/package.json
@@ -1,15 +1,25 @@
 {
-  "name": "testsite.test",
-  "version": "1.0.0",
-  "description": "",
+  "name": "portfolio",
+  "version": "1.0.1",
+  "description": "Portfolio Project",
+  "author": "John Fasano",
   "main": "index.js",
+  "license": "ISC",
+  "engines": {
+    "node": "8.15.0",
+    "npm": "6.4.1"
+  },
+  "os": [
+    "ubuntu",
+    "linux",
+    "windows",
+    "macOS"
+  ],
   "scripts": {
     "dev": "webpack-dev-server --env=dev --mode development --open",
     "start": "webpack --env=prod",
     "test": "jest --passWithNoTests"
   },
-  "author": "",
-  "license": "ISC",
   "devDependencies": {
     "@babel/cli": "^7.4.4",
     "@babel/core": "^7.4.4",

